pkg/storegateway/bucket_stores.go

```diff
@@ -464,6 +464,10 @@ func (u *BucketStores) getOrCreateStore(userID string) (*BucketStore, error) {
 		bucketStoreOpts = append(bucketStoreOpts, WithDebugLogging())
 	}
 
+	if u.limits.IgnoreNativeHistogramsOnRead(userID) {
+		bucketStoreOpts = append(bucketStoreOpts, WithIgnoreNativeHistograms())
+	}
```

Comment:
This line will get called only on startup and when new tenants appear during execution. This means that if the runtime config changes after we've already created a store for a tenant in `getOrCreateStore`, the created store will not get the new value in its `ignoreNativeHistograms`. 

Another option is to read the overrides for each request and pass the boolean into `synchronousSeriesSet` and `streamingSeriesSetForBlocks`. To do that we will need access to the overrides from a `BucketStore` (store-gateway of a single tenant). The `BucketStores` (holding per-tenant store-gateways) already has the overrides ([code](https://github.com/grafana/mimir/blob/496ee6fc7650273531d753dd11e64cc11f18a288/pkg/storegateway/bucket_stores.go#L476)). You can pass them to `BucketStore` via a `WithTenantOverrides(*validation.Overrides)`.

Alternatively to holding all overrides for all tenants, you can pass an interface or a function which returns a boolean and use that function for each incoming request in `synchronousSeriesSet` and `streamingSeriesSetForBlocks`. I think I prefer this approach, but go with whatever feels better to you.

Reply:
Thanks for pointing that out! Your suggestion sounds good--I'll get it implemented!